include/power_grid_model/math_solver/iterative_pf_solver.hpp

```diff
@@ -121,6 +123,7 @@ class IterativePFSolver {
                 ComplexTensor<sym> const y_ref = y_bus.math_model_param().source_param[source];
                 output.source[source].i = dot(y_ref, u_ref - output.u[bus]);
                 output.source[source].s = output.u[bus] * conj(output.source[source].i);
+                output.bus_injection[bus] += output.source[source].s;
```

Comment:
Please calculate bus injection with the function `calculate_injection` from `YBus`